### ISSUE TITLE
#1 Setup of the development stack

### ISSUE THREAD

**Issue by rcavaliere** [noi-techpark/opendatahub-analytics-v2#1]
- [ ] Setup of the Vue 3 project
- [ ] Export and inclusion of core components from the Open Data Hub portal
- [ ] Creation of a responsive navigation structure.

Task for @rcavaliere 
- [ ] Verify with @sseppi unique package to be used as web-component for the UI of different but similar Open Data Hub applications

**Comment by rcavaliere** [comment]
@MatteoBiasi I discussed with @sseppi  your request, but actually we should better understand this. Can you provide more details about this web-component should provide?

**Comment by MatteoBiasi** [comment]
Hello everyone, as I was explaining to @rcavaliere, our idea is to package and include the current base components of the Open Data Browser as web components, so we can maintain a consistent design. Specifically, we are thinking of:
- Header
- Buttons
- Box
- Select input with search
- Checkbox
- Map component (to be evaluated)

From a technical perspective, this would involve configuring these components to function as web components, exporting them to an npm repository (or similar), and then importing them into the analytics tool project.

Let me know your thoughts!

**Comment by rcavaliere** [comment]
@MatteoBiasi as just discussed, please go and implement this. We however decided to not include the map, as this could be application specific. 

@clezag - there would be the need to create a central reference point, e.g. a NPM registry that is the possible to integrate in order to use these shared component. If I got it well, we then need:

- a dedicated repository in which we store the source code of such webcomponents
- a NPM registry or similar for the easy integration of such components in Open Data Hub applications.

Can you please set up this please, possibly by the beginning of next week? In this way it is then possible to proceed with the implementation of the new version of the analytics tool.

**Comment by sseppi** [comment]
@MatteoBiasi during the implementation of the Data Browser we already tested and implemented some components that could be converted. We also experienced some issues, therefore if you want we are available to share our experience, to speedup the development as much as possible.

**Comment by MatteoBiasi** [comment]
@sseppi sure. Maybe we can have a quick talk during Thursday's Open Data Browser meeting?

**Comment by clezag** [comment]
@rcavaliere @MatteoBiasi 
I will create a git repo to host the components. 

How about just publishing the components on the public NPM registry? 
This would be in line with our Open Source first policy, and we avoid a lot of extra work in setting up something ourselves.

**Comment by rcavaliere** [comment]
@clezag yes, I don't see issues with that. Let's go public!

**Comment by sseppi** [comment]
@clezag I agree, and in future someone could also reuse our components and/or contribute to optimize them.

**Comment by clezag** [comment]
I've setup [this repo](https://github.com/noi-techpark/opendatahub-ui-components) to host the webcomponents. Let me know if you need anything else.

We don't currently publish on npm, so I would kindly ask to also provide the necessary CI/CD as part of the PR.

**Comment by MatteoBiasi** [comment]
Thanks @clezag. We well start to work on this part during the next days and come to you with further details about the CI/CD.
